src/wp-includes/media.php

```diff
@@ -1137,6 +1137,16 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 			}
 		}
 
+		// Adds 'auto' to the sizes attribute if applicable.
+		if (
+			isset( $attr['loading'] ) &&
+			'lazy' === $attr['loading'] &&
+			isset( $attr['sizes'] ) &&
+			! wp_sizes_attribute_includes_valid_auto( $attr['sizes'] )
+		) {
+			$attr['sizes'] = 'auto, ' . $attr['sizes'];
```

Comment:
Should this _remove_ `auto` if it is present in `$attr['sizes']` while `$attr['loading']` is not `lazy`? This is implemented in https://github.com/WordPress/performance/pull/1476.

Reply:
> Should this _remove_ `auto` if it is present in `$attr['sizes']` while `$attr['loading']` is not `lazy`? This is implemented in [WordPress/performance#1476](https://github.com/WordPress/performance/pull/1476).

Do any of the implementations respect it in spite of the spec if the CSS is available and the image isn't set for lazy loading? If so then I think it would be good to keep it given the sizes attributes can be inaccurate in many situations and the CSS may be in the HTML header and therefore available.

Reply:
> Should this _remove_ `auto` if it is present in `$attr['sizes']` while `$attr['loading']` is not `lazy`? This is implemented in [WordPress/performance#1476](https://github.com/WordPress/performance/pull/1476).

@westonruter Core only adds the "auto" in sizes when image is lazy-loaded so if user added that through filter then core will not remove "auto".

@peterwilsoncc Even if the CSS is available early in the document (such as in the HTML header), implementations do not dynamically adjust the sizes attribute based on CSS unless lazy loading is enabled.

Reply:
I don't think this is needed as Core doesn't add `sizes="auto"` in a way where this could be relevant. If a plugin does that, it's doing it wrong.

Now I'm not strongly against adding this, but it's not a requirement for this feature. We could discuss whether this is worth adding in a separate ticket, or wait if it comes up as a real use-case problem somewhere.